File: src/lessons/bat/string1/StringSplosionEntity.java
package lessons.bat.string1;

import static plm.core.ValueSerializer.*;

import plm.universe.bat.BatEntity;

public class StringSplosionEntity extends BatEntity {

  public void run()
  {
    int count = getTestCount();
    for (int i = 0; i < count; i++) {
      Object[] param = (Object[])deserialize(getTest(i));
      setTestResult(i, serialize(stringSplosion((String)param[0])));
    }
  }

  /* BEGIN TEMPLATE */
  String stringSplosion(String str)
  {
    /* BEGIN SOLUTION */
    String result = "";
    // On each iteration, add the substring of the chars 0..i
    for (int i = 0; i < str.length(); i++) {
      result = result + str.substring(0, i + 1);
    }
    return result;
    /* END SOLUTION */
  }
  /* END TEMPLATE */
}

File: src/lessons/bat/string1/StringSplosionEntity.scala
package lessons.bat.string1

import plm.core.ValueSerializer
import plm.universe.bat.BatEntity

class StringSplosionEntity extends BatEntity {

    override def run() {
      import ValueSerializer._

      val count = getTestCount()
      for (i <- 0 to count -1) {
        val param = deserialize(getTest(i)).asInstanceOf[Array[Object]]
        setTestResult(i, serialize(stringSplosion(param(0).asInstanceOf[String])) )
	  }
	}

	/* BEGIN TEMPLATE */
	def stringSplosion(str:String):String = {
		/* BEGIN SOLUTION */
		var res = ""
		for (i <- 0 to str.length-1) 
			res ++= str.substring(0,i+1)
		return res
		/* END SOLUTION */
	}
	/* END TEMPLATE */
}
